daemon/snap.go

```diff
@@ -183,19 +179,24 @@ func mapLocal(about aboutSnap) map[string]interface{} {
 		appNames = append(appNames, appName)
 	}
 	sort.Strings(appNames)
-	apps := make([]appJSON, 0, len(localSnap.Apps))
-	for _, appName := range appNames {
+	apps := make([]*client.AppInfo, len(localSnap.Apps))
+	// TODO: pass in an actual notifier here instead of nil
+	//       (Status doesn't _need_ it, but benefits from it)
+	sysd := systemd.New(dirs.GlobalRootDir, nil)
+	for i, appName := range appNames {
 		app := localSnap.Apps[appName]
-		var installedDesktopFile string
-		if osutil.FileExists(app.DesktopFile()) {
-			installedDesktopFile = app.DesktopFile()
+		apps[i] = &client.AppInfo{Name: app.Name}
+		if fn := app.DesktopFile(); osutil.FileExists(fn) {
+			apps[i].DesktopFile = fn
 		}
 
-		apps = append(apps, appJSON{
-			Name:        app.Name,
-			Daemon:      app.Daemon,
-			DesktopFile: installedDesktopFile,
-		})
+		if app.IsService() {
+			// TODO: look into making a single call to Status for all services
+			sts, err := sysd.Status(app.ServiceName())
+			if err == nil && len(sts) == 1 {
```

Comment:
Yeah, not sure it makes sense to be otherwise for this. For the services endpoint itself this would be a failure.